ui/src/app/applications/components/pod-logs-viewer/pod-logs-highlight-button.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import {useState, useRef, useEffect} from 'react';
+import {ToggleButton} from '../../../shared/components/toggle-button';
+
+// PodHighlightButton is a component that renders a toggle button that toggles pod highlighting.
+export const PodHighlightButton = ({selectedPod, setSelectedPod, pods}: {selectedPod: string | null; setSelectedPod: (value: string | null) => void; pods: string[]}) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const dropdownRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        const handleClickOutside = (event: MouseEvent) => {
+            if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+                setIsOpen(false);
+            }
+        };
+
+        document.addEventListener('mousedown', handleClickOutside);
+        return () => document.removeEventListener('mousedown', handleClickOutside);
+    }, []);
+
+    return (
+        <div ref={dropdownRef} style={{position: 'relative'}}>
+            <ToggleButton title='Select a pod to highlight its logs' onToggle={() => setIsOpen(!isOpen)} icon='highlighter' toggled={selectedPod !== null} />
+            {isOpen && (
+                <div className='select-container' style={{position: 'absolute', top: '100%', left: 0, zIndex: 1}}>
+                    <div className='select-options'>
+                        {pods.map(pod => (
```

Comment:
Does the background or foreground colour need to be changed for dark or light mode?

Reply:
Could it be fixed as part of this PR, since this is for the new dropdown?  (The changes in pod-logs-viewer, also considers the dark mode setting too.)

<img width="526" alt="contrastInDarkMode" src="https://github.com/user-attachments/assets/68492788-b6cb-4a75-bd49-72dca6cde84b" />

Reply:
There are a whole bunch of colors that are broken. I worry it's a boil the ocean problem. There appear to be at ~5 different "blacks" plus various "whites" and "cyans". I wouldn't even know which colors to suggest.

Offhand, `containing` shown here is too dark and needs to be lightened. The log pane itself could be true black or one of the various other blacks. As for the text of the log itself, maybe it should be pure white to match the `LOGS` tab color?

Reply:
@aali309 those colors might be ok in dark, they're definitely wrong for light mode.

Reply:
Ok, I figured out what's going on... it isn't a related bug, it's just a really silly interaction:
- https://github.com/argoproj/argo-cd/issues/21426

Some of the problems @keithchong is describing remind me of:
- https://github.com/argoproj/argo-cd/issues/13907